### PR TITLE
Document EnvironmentEDFs.

### PR DESCRIPTION
**Pull request by est77**
Part of issue #745.
Also refactored sky dome models input metadata.

Suggestions for missing doc strings and improvements welcome.

### CODE REVIEW

File: src/appleseed/renderer/modeling/environmentedf/constanthemisphereenvironmentedf.cpp [hunk 1]
```diff
@@ -184,7 +184,8 @@ Dictionary ConstantHemisphereEnvironmentEDFFactory::get_model_metadata() const
     return
         Dictionary()
             .insert("name", Model)
-            .insert("label", "Per-Hemisphere Constant Environment EDF");
+            .insert("label", "Per-Hemisphere Constant Environment EDF")
+            .insert("help", "Per-hemisphere constant environment");
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/constanthemisphereenvironmentedf.cpp]: An environment made of two hemispheres of constant radiance.

File: src/appleseed/renderer/modeling/environmentedf/gradientenvironmentedf.cpp [hunk 2]
```diff
@@ -191,7 +191,8 @@ Dictionary GradientEnvironmentEDFFactory::get_model_metadata() const
     return
         Dictionary()
             .insert("name", Model)
-            .insert("label", "Gradient Environment EDF");
+            .insert("label", "Gradient Environment EDF")
+            .insert("help", "Gradient environment");
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/gradientenvironmentedf.cpp]: An environment made of a vertical radiance gradient, from zenith to horizon.

File: src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "sun_theta")
+            .insert("label", "Sun Theta Angle")
+            .insert("type", "numeric")
+            .insert("min_value", "0.0")
+            .insert("max_value", "90.0")
+            .insert("use", "required")
+            .insert("default", "45.0")
+            .insert("help", "Sun polar angle in degrees"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp]: "Sun polar (vertical) angle in degrees"

File: src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "sun_phi")
+            .insert("label", "Sun Phi Angle")
+            .insert("type", "numeric")
+            .insert("min_value", "-360.0")
+            .insert("max_value", "360.0")
+            .insert("use", "required")
+            .insert("default", "0.0")
+            .insert("help", "Sun azimuthal angle in degrees"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp]: "Sun azimuthal (horizontal) angle in degrees"

File: src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "luminance_multiplier")
+            .insert("label", "Luminance Multiplier")
+            .insert("type", "numeric")
+            .insert("min_value", "0.0")
+            .insert("max_value", "10.0")
+            .insert("use", "optional")
+            .insert("default", "1.0")
+            .insert("help", "TODO: Document me"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp]: "Sky luminance multiplier"

File: src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "luminance_gamma")
+            .insert("label", "Luminance Gamma")
+            .insert("type", "numeric")
+            .insert("min_value", "0.0")
+            .insert("max_value", "3.0")
+            .insert("use", "optional")
+            .insert("default", "1.0")
+            .insert("help", "TODO: Document me"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp]: "Sky luminance gamma"

File: src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "saturation_multiplier")
+            .insert("label", "Saturation Multiplier")
+            .insert("type", "numeric")
+            .insert("min_value", "0.0")
+            .insert("max_value", "10.0")
+            .insert("use", "optional")
+            .insert("default", "1.0")
+            .insert("help", "TODO: Document me"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp]: "Sky color saturation multiplier"

File: src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "horizon_shift")
+            .insert("label", "Horizon Shift")
+            .insert("type", "text")
+            .insert("use", "optional")
+            .insert("default", "0.0")
+            .insert("help", "TODO: Document me"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/ienvironmentedffactory.cpp]: "Shift the sky horizontally by an amount specified in degrees"

File: src/appleseed/renderer/modeling/environmentedf/hosekenvironmentedf.cpp [hunk 9]
```diff
@@ -513,45 +476,8 @@ DictionaryArray HosekEnvironmentEDFFactory::get_input_metadata() const
...
-
-    metadata.push_back(
-        Dictionary()
-            .insert("name", "horizon_shift")
-            .insert("label", "Horizon Shift")
-            .insert("type", "text")
-            .insert("use", "optional")
-            .insert("default", "0.0"));
+            .insert("default", "0.3")
+            .insert("help", "TODO: Document me"));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/hosekenvironmentedf.cpp]: "Ground albedo (reflection coefficient of the ground)"

File: src/appleseed/renderer/modeling/environmentedf/latlongmapenvironmentedf.cpp [hunk 10]
```diff
@@ -423,7 +423,8 @@ Dictionary LatLongMapEnvironmentEDFFactory::get_model_metadata() const
     return
         Dictionary()
             .insert("name", Model)
-            .insert("label", "Latitude-Longitude Map Environment EDF");
+            .insert("label", "Latitude-Longitude Map Environment EDF")
+            .insert("help", "Lat-Long image based lighting environment");
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/latlongmapenvironmentedf.cpp]: "image-based"

File: src/appleseed/renderer/modeling/environmentedf/mirrorballmapenvironmentedf.cpp [hunk 11]
```diff
@@ -190,7 +190,8 @@ Dictionary MirrorBallMapEnvironmentEDFFactory::get_model_metadata() const
     return
         Dictionary()
             .insert("name", Model)
-            .insert("label", "Mirror Ball Map Environment EDF");
+            .insert("label", "Mirror Ball Map Environment EDF")
+            .insert("help", "Mirror ball image based lighting environment");
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/environmentedf/mirrorballmapenvironmentedf.cpp]: "image-based"

**Comment by est77** [comment]
Fixed and rebased.

**Comment by dictoon** [comment]
Thanks!

**Pull request merged**